### PR TITLE
Skip formatting if the language flavor shows this is not an MSSQL doc

### PR DESCRIPTION
**Pull request by kevcunnane**

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs [hunk 1]
```diff
@@ -103,6 +110,14 @@ private static TelemetryProperties CreateTelemetryProps(bool isDocFormat)
...
         {
+            if (docFormatParams == null
+                || docFormatParams.TextDocument == null
+                || docFormatParams.TextDocument.Uri == null
+                || ShouldSkipFormatting(docFormatParams.TextDocument.Uri))
+            {
+                return Array.Empty<TextEdit>();
+            }
+
             return await Task.Factory.StartNew(() =>
```
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs]: What's the benefit of starting up a new thread for something that returns immediately? Why not do a `Task.FromResult()` sort of thing?

File: src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs [hunk 2]
```diff
@@ -103,6 +110,14 @@ private static TelemetryProperties CreateTelemetryProps(bool isDocFormat)
 
         private async Task<TextEdit[]> FormatRangeAndReturnEdits(DocumentRangeFormattingParams docFormatParams)
         {
+            if (docFormatParams == null
+                || docFormatParams.TextDocument == null
+                || docFormatParams.TextDocument.Uri == null
+                || ShouldSkipFormatting(docFormatParams.TextDocument.Uri))
+            {
+                return Array.Empty<TextEdit>();
```
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs]: I'm not very familiar with this way of instantiating an empty array. Can't you do it with empty `[]`?

Comment (kevcunnane) [reply src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs]: You can do it with `new TextEdit[0]`. I just prefer this way as it has nice clarity in my opinion.

File: src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs (hunk 1 above)
Comment (kevcunnane) [reply src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs]: I'll put the other code inside the taskfactory, but the purpose is that these are event handlers for message channel notifications. Per previous discussions we don't want to hold these up - formatting the code is usually fast but for a large file could take a second. Why hold the channel up for that?

**Pull request merged**